ethcore/src/state/mod.rs

```diff
@@ -401,9 +401,12 @@ impl<B: Backend> State<B> {
 		self.factories.vm.clone()
 	}
 
-	/// Create a recoverable checkpoint of this state.
-	pub fn checkpoint(&mut self) {
-		self.checkpoints.get_mut().push(HashMap::new());
+	/// Create a recoverable checkpoint of this state. Return the checkpoint index.
+	pub fn checkpoint(&mut self) -> usize {
```

Comment:
I'm a bit hesitant on exposing the index here. If we merge / discard the checkpoint the index might point to a completely different one or be plain invalid.

Reply:
Yes you're right. I'm planning to use #9427 (where we can get better references to checkpoints). However, right now I can't think of any better ways.

Reply:
Makes sense!